### ISSUE TITLE
[issue]: Ventoy BootPwd Not Working

### ISSUE THREAD

**Issue by SpikeTurbo** [ventoy/Ventoy#2584]
### Official FAQ

- [X] I have checked the official FAQ.

### Ventoy Version

1.0.95

### What about latest release

Yes. I have tried the latest release, but the bug still exist.

### Try alternative boot mode

Yes. I have tried them, but the bug still exist.

### BIOS Mode

Both

### Partition Style

MBR

### Disk Capacity

64GB

### Disk Manufacturer

Lexar

### Image file checksum (if applicable)

None

### What happened?

I tried to modify my USB to include a password each time Ventoy starts using VentoyPlugson, however when I go to edit the configuration, it does not save the password I apply. The menupwd options work fine and save as expected, but the others do not. I've tested this in UEFI and Legacy modes, but nothing works.

I also attempted to do this using a text editor (Notepad++), but kept getting syntax errors.

![image](https://github.com/ventoy/Ventoy/assets/24995553/9b8277e4-7cd6-4a0f-820b-2530074d46fe)

**Comment by ventoy** [comment]
1. When you set `bootpwd` did it popup the success message? What is the content of ventoy.json after that? Can you upload it here?
2. When you set any other password, click the `preview`  button in the top of the page to check the content.

**Comment by noigamegun** [comment]
> 1. When you set `bootpwd` did it popup the success message? What is the content of ventoy.json after that? Can you upload it here?
> 2. When you set any other password, click the `preview`  button in the top of the page to check the content.

I have the same problem.
it seem like there is no changes in the .json file even in the preview.

**Comment by SpikeTurbo** [comment]
> 1. When you set `bootpwd` did it popup the success message? What is the content of ventoy.json after that? Can you upload it here?
> 
>     2. When you set any other password, click the `preview`  button in the top of the page to check the content.

@noigamegun is correct. There are no changes made to the JSON file even after checking the Preview.

However, when I set a menu password for a specific ISO, I get this added to the JSON file:

```
"password":{
        "menupwd":[
            {
                "file": "/Win10_22H2.iso",
                "pwd": "md5#2e814b294cfff6b70bfc25b789366398"
            }
        ]
    },
```

The only thing that works in the `menupwd` variable. `bootpwd`, `isopwd`, `wimpwd`, `imgpwd`, `vhdpwd`, `efipwd` and `vtoypwd` can be edited, but the changes are not written to the JSON file.

As requested @ventoy, here is my JSON file output according to the Preview after attempting to apply the `bootpwd` variable:

```
{
    "control":[
        { "VTOY_SECONDARY_BOOT_MENU": "0" },
        { "VTOY_MENU_TIMEOUT": "15" }
    ],
    "theme":{
        "file": "/ventoy/themes/cpcc2/theme.txt",
        "ventoy_color": "#212121"
    },
    "menu_alias_legacy":[
        {
            "image": "/dban-2.3.0_i586.iso",
            "alias": "DBAN"
        },
        {
            "image": "/HBCD_PE_x64.iso",
            "alias": "HBCD"
        },
        {
            "image": "/linuxmint-21.2-cinnamon-64bit.iso",
            "alias": "Linux Mint (21.2)"
        }
    ],
    "menu_alias_uefi":[
        {
            "image": "/linuxmint-21.2-cinnamon-64bit.iso",
            "alias": "Linux Mint (21.2)"
        },
        {
            "image": "/Win10_22H2.iso",
            "alias": "Windows 10 (22H2)"
        },
        {
            "image": "/Win11_22H2.iso",
            "alias": "Windows 11 (22H2)"
        }
    ],
    "menu_tip_legacy":{
        "top": "71%",
        "color": "white",
        "tips":[
            {
                "image": "/HBCD_PE_x64.iso",
                "tip": "Hiren's Boot CD (Disconnect from Network before using)"
            },
            {
                "image": "/dban-2.3.0_i586.iso",
                "tip": "Erases the internal HDD (will not work on SSDs)"
            },
            {
                "image": "/linuxmint-21.2-cinnamon-64bit.iso",
                "tip": "Boot into Linux Mint"
            }
        ]
    },
    "menu_tip_uefi":{
        "top": "71%",
        "color": "white",
        "tips":[
            {
                "image": "/linuxmint-21.2-cinnamon-64bit.iso",
                "tip": "Boot into Linux Mint"
            },
            {
                "image": "/Win10_22H2.iso",
                "tip": "Install Windows 10 (OEM)"
            },
            {
                "image": "/Win11_22H2.iso",
                "tip": "Install Windows 11 (OEM)"
            }
        ]
    },
    "menu_class_legacy":[
        {
            "key": "dban",
            "class": "siduction"
        },
        {
            "key": "HBCD",
            "class": "kali"
        },
        {
            "key": "linuxmint",
            "class": "linuxmint"
        }
    ],
    "menu_class_uefi":[
        {
            "key": "linuxmint",
            "class": "linuxmint"
        },
        {
            "key": "Win10",
            "class": "windows"
        },
        {
            "key": "Win11",
            "class": "windows11"
        }
    ],
    "persistence":[
        {
            "image": "/linuxmint-21.2-cinnamon-64bit.iso",
            "backend":[
                "/ventoy/data/lmint.dat"
            ]
        }
    ],
    "conf_replace_legacy":[
        {
            "iso": "/dban-2.3.0_i586.iso",
            "org": "/isolinux.cfg",
            "new": "/ventoy/dban_config.cfg"
        }
    ],
    "image_list_legacy":[
        "/dban-2.3.0_i586.iso",
        "/HBCD_PE_x64.iso",
        "/linuxmint-21.2-cinnamon-64bit.iso"
    ],
    "image_list_uefi":[
        "/linuxmint-21.2-cinnamon-64bit.iso",
        "/Win10_22H2.iso",
        "/Win11_22H2.iso"
    ]
}
```

**Comment by noigamegun** [comment]
> I also attempted to do this using a text editor (Notepad++), but kept getting syntax errors.

Yeah tried that. just getting syntax error.
Maybe try putting the "syntax error" .json file and boot it?
If the `bootpwd` screen don't appear.  Maybe check the plugin settings in the f5 menu if it found the `bootpwd` or not

**Comment by noigamegun** [comment]
@ventoy I submitted a success image report. Feel free to check it. The chromeos one.

**Comment by SpikeTurbo** [comment]
> > I also attempted to do this using a text editor (Notepad++), but kept getting syntax errors.
> 
> Yeah tried that. just getting syntax error.
> Maybe try putting the "syntax error" .json file and boot it?
> If the `bootpwd` screen don't appear.  Maybe check the plugin settings in the f5 menu if it found the `bootpwd` or not
> 

Did you get it to work by doing this? Can you explain what you mean?

**Comment by SpikeTurbo** [comment]
@ventoy I see you modified your code. Do I need to recreate my Ventoy drive or will you release a new version that corrects this bug?

**Comment by ventoy** [comment]
@SpikeTurbo @noigamegun 
Test this CI release 
https://github.com/ventoy/Ventoy/actions/runs/6395504951
https://www.ventoy.net/en/doc_github_ci.html

You don't need to recreate your Ventoy drive, just use the VentoyPlugson in the CI release.

**Comment by SpikeTurbo** [comment]
> @SpikeTurbo @noigamegun Test this CI release https://github.com/ventoy/Ventoy/actions/runs/6395504951 https://www.ventoy.net/en/doc_github_ci.html
> 
> You don't need to recreate your Ventoy drive, just use the VentoyPlugson in the CI release.

The CI release listed appears to have fixed this issue. Will this be included as a fix in the next version?

**Comment by ventoy** [comment]
Yes, this will be included in the next release.

**Comment by ventoy** [comment]
This has been fixed in the 1.0.96 release.
